src/USER-MEAMC/pair_meamc.cpp

```diff
@@ -358,272 +351,214 @@ double PairMEAMC::init_one(int i, int j)
+  MPI_Bcast(t3.data(), nelements, MPI_DOUBLE, 0, world);
+  MPI_Bcast(rozero.data(), nelements, MPI_DOUBLE, 0, world);
 
   // pass element parameters to MEAM package
 
-  meam_inst->meam_setup_global(nelements,lat,ielement,atwt,alpha,b0,b1,b2,b3,
-                       alat,esub,asub,t0,t1,t2,t3,rozero,ibar);
+  meam_inst->meam_setup_global(nelements, lat.data(), ielement.data(), atwt.data(),
+                               alpha.data(), b0.data(), b1.data(), b2.data(), b3.data(),
+                               alat.data(), esub.data(), asub.data(), t0.data(), t1.data(),
+                               t2.data(), t3.data(), rozero.data(), ibar.data());
 
   // set element masses
 
   for (int i = 0; i < nelements; i++) mass[i] = atwt[i];
+}
 
-  // clean-up memory
-
-  delete [] lat;
-  delete [] ielement;
-  delete [] ibar;
-  delete [] z;
-  delete [] atwt;
-  delete [] alpha;
-  delete [] b0;
-  delete [] b1;
-  delete [] b2;
-  delete [] b3;
-  delete [] alat;
-  delete [] esub;
-  delete [] asub;
-  delete [] t0;
-  delete [] t1;
-  delete [] t2;
-  delete [] t3;
-  delete [] rozero;
-  delete [] found;
+/* ---------------------------------------------------------------------- */
 
+void PairMEAMC::read_user_meamc_file(const std::string &userfile)
+{
   // done if user param file is "NULL"
 
   if (userfile == "NULL") return;
 
   // open user param file on proc 0
 
+  std::shared_ptr<PotentialFileReader> reader;
+
   if (comm->me == 0) {
-    fp = utils::open_potential(userfile,lmp,nullptr);
-    if (fp == nullptr)
-      error->one(FLERR,fmt::format("Cannot open MEAM potential file {}",
-                                   userfile));
+    reader = std::make_shared<PotentialFileReader>(lmp, userfile, "MEAM");
   }
 
   // read settings
   // pass them one at a time to MEAM package
   // match strings to list of corresponding ints
+  char * line = nullptr;
+  char buffer[MAXLINE];
 
-  int maxparams = 6;
-  char **params = new char*[maxparams];
   while (1) {
     int which;
     int nindex, index[3];
     double value;
-    char line[MAXLINE];
     int nline;
-    char *ptr;
     if (comm->me == 0) {
-      ptr = fgets(line,MAXLINE,fp);
-      if (ptr == nullptr) {
-        fclose(fp);
+      line = reader->next_line();
+      if (line == nullptr) {
         nline = -1;
       } else nline = strlen(line) + 1;
+    } else {
+      line = buffer;
     }
+
     MPI_Bcast(&nline,1,MPI_INT,0,world);
     if (nline<0) break;
     MPI_Bcast(line,nline,MPI_CHAR,0,world);
 
-    // strip comment, skip line if blank
-
-    if ((ptr = strchr(line,'#'))) *ptr = '\0';
-    if (utils::count_words(line) == 0) continue;
-
-    // params = ptrs to all fields in line
-
-    int nparams = 0;
-    params[nparams++] = strtok(line,"=(), '\t\n\r\f");
-    while (nparams < maxparams &&
-           (params[nparams++] = strtok(nullptr,"=(), '\t\n\r\f")))
-      continue;
-    nparams--;
+    ValueTokenizer values(line, "=(), '\t\n\r\f");
```

Comment:
Damn, I had completely forgotten how hacky this namelist "parsing" was...
(just a random thought, it's fine for this PR)